packages/mantine/src/components/table/table-actions/TableActions.tsx

```diff
@@ -1,11 +1,14 @@
-import {Factory, Grid, GridColProps, Group, useProps} from '@mantine/core';
-import {ForwardedRef, ReactElement, ReactNode} from 'react';
+import {MoreSize16Px} from '@coveord/plasma-react-icons';
+import {ActionIcon, Button, Factory, GridColProps, Menu, Tooltip, useProps} from '@mantine/core';
+import {Children, ForwardedRef, ReactElement, ReactNode} from 'react';
 
 import {CustomComponentThemeExtend, identity} from '../../../utils';
-import {TableComponentsOrder} from '../Table';
+import {Table} from '../Table';
 import {useTableContext} from '../TableContext';
 
-export type TableActionsStylesNames = 'actionsRoot' | 'actionsGroup';
+export type TableActionsStylesNames = 'actionsTarget' | 'actionsDropdown' | 'actionsTooltip';
+
+export type TableActionsItems = typeof Table.ActionItem | typeof Table.ActionLabel | typeof Table.ActionDivider;
```

Comment:
We could change the type here like this

```ts
export type TableActionsItem = typeof Table.ActionItem | typeof Table.ActionLabel | typeof Table.ActionDivider;
export type TableActionsItems = TableActionsItem | Iterable<TableActionsItem>;

children: ((datum: T) => TableActionsItems) | ((data: T[]) => TableActionsItems);
```

This way it would allow setting actions using a fragment instead of the array.

```tsx
{(selectedRow: Person) => (
    <>
        <Table.ActionItem
            primary
            onClick={() => alert(`Action 1 triggered for row: ${selectedRow.id}`)}
            leftSection={<EditSize16Px height={16} />}
        >
            Action 1
        </Table.ActionItem>
        <Table.ActionItem
            primary
            onClick={() => alert(`Action 2 triggered for row: ${selectedRow.id}`)}
            leftSection={<AbTestingSize16Px height={16} />}
        >
            Action 2
        </Table.ActionItem>
    </>
)}
```

The type also accepts a single action without a fragment.

```tsx
{(selectedRow: Person) => (
    <Table.ActionItem
        primary
        onClick={() => alert(`Action 1 triggered for row: ${selectedRow.id}`)}
        leftSection={<EditSize16Px height={16} />}
    >
        Action 1
    </Table.ActionItem>
)}
```

I don't know how complicated it is to support in the table itself thought. I think Children.toArray() already supports this.

Reply:
That would be really nice, I'm gonna try it out! (I didn't really like the array, it felt against React's way of doing things)